### PR TITLE
Updating npm to 0.5

### PR DESCRIPTION
**Pull request by donny-dont**
Updating the npm plugin to 0.5 format.

Will start working on documentation for usage. Comments welcome @bradrydzewski

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,98 +6,52 @@
...
-    },
-    "vargs": {
-        "username": "octocat",
-        "password": "pa$$word",
-        "email": "octocat@github.com"
-    }
-}
-EOF
+docker run --rm \
+  -e NPM_USERNAME=drone \
```
Comment (bradrydzewski) [inline README.md]: maybe add an example for mounting the current volume?

```
  -v $(pwd):$(pwd) \
  -w $(pwd) \
```

Comment (donny-dont) [inline README.md]: Think I beat you to that :stuck_out_tongue_winking_eye:

**Comment by donny-dont** [comment]
@bradrydzewski I think I'm happy with it if you want to do a review.

**Comment by nlf** [comment]
this looks good :+1:

**Comment by donny-dont** [comment]
@nlf what do you think about using `npm info` rather than making the request out?

**Comment by nlf** [comment]
hmm, that may be a better idea since it would then inherit the auth that was setup in the npmrc file (which it appears it currently does not auth that request?)

**Comment by donny-dont** [comment]
Yea the only thing I'm concerned about is how to communicate back what happened.

**Comment by tboerger** [comment]
LGTM

File: README.md [hunk 2]
```diff
@@ -6,98 +6,56 @@
...
+```
 
-Build the container using `make`:
+Please note incorrectly building the image for the correct x64 linux and with GCO disabled will result in an error when running the Docker image:
 
 ```
-make deps docker
+docker: Error response from daemon: Container command
+'/bin/drone-git' not found or does not exist..
+```
```
Comment (bradrydzewski) [inline README.md]: change to `drone-npm` :)

**Comment by bradrydzewski** [comment]
LGTM

> what do you think about using npm info pkg@version version rather than making the request out?

Let me know what you decide and if you want to adjust the current approach, or if we should merge this as-is.

**Comment by donny-dont** [comment]
@bradrydzewski @tboerger @nlf Just reworked the plugin. I tested it with NPM enterprise and Sinopia locally and it seems to be working.

**Comment by bradrydzewski** [comment]
LGTM 

I'm guessing build is failing because we need to vendor dependencies? I can do this when I update the secrets locally.

**Comment by donny-dont** [comment]
:confetti_ball:

**Pull request merged**